specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-09-01/virtualMachine.json

```diff
@@ -1891,6 +1891,331 @@
           "final-state-via": "location"
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}/VMApplications": {
```

Comment:
>subscriptions [](http://example.com/codeflow?start=6&length=13)

The response of the LIST operation VirtualMachineApplicationsProxyResourceListResult is a wrapper on VMApplicationProxyResource is a containment of VMGalleryApplication

The response of the point GET operation 
VMApplicationProxyResourceWithInstanceView is a VMGalleryApplicationWithInstanceView is a containment of the union of instanceView, VMGalleryApplication


Point get has more properties than the response of the LIST. This is not allowed. You need both of these responses to match

Reply:
I see the difference is because you want to carry forward the existing pattern of $expand. Generally speaking its an anti-pattern to have such a usage where the payload varies based on query params and makes it difficult for customers to code their deserialization logic. However, since you already have this pattern in other resources, we have to let this go